pdfs/tests/Main.hs

```diff
@@ -0,0 +1,50 @@
+
+import Test.QuickCheck (property, Arbitrary (..))
+import Data.ByteString.Char8 as LBS
+
+import Pdf.Scanner
+
+import Test.Hspec (hspec, describe, it)
+
+instance Arbitrary LBS.ByteString where
+    arbitrary = LBS.pack `fmap` arbitrary
+
+main :: IO ()
+main = hspec $ do
+    let inf  = MaxSearchLength Nothing
+        zero = MaxSearchLength (Just 0)
+
+    describe "injection/projection works" $ do
+        it "project _ x (inject _ x text) == x" . property $
+            \(before, after, piece) ->
+                let
+                    text = before <> insertionMark <> after
```

Comment:
I don't think there's anything wrong with having `let` definitions on the same line as `let` itself...

Reply:
Yeah, its just looked as a wall of text in the moment.